src/app/file-manager/file-download-modal/file-download-modal.component.ts

```diff
@@ -76,27 +76,9 @@ export class FileDownloadModalComponent  {
   setItems() {
     this.data.selectedFiles.filter(file => file.type === 'File')
       .forEach(file => {
-        const dataFile = {
-          id: file.id,
-          name: file.name,
-          size: file.size,
-          created: file.updated,
-          updated: file.updated,
-          icon: file.type === 'Folder' ? 'folder' : 'cloud',
-          selected: false,
-          destination: this.directory,
-          preserveStructure: this.preserveStructure,
-          mediaLink: file.mediaLink,
-          path: file.path,
-          progress: 0,
-          type: Type.DOWNLOAD,
-          status: ItemStatus.PENDING,
-          transferred: 0,
-          remaining: 0,
-          bucketName: '',
-          delimiter: '',
-          prefix: ''
-        };
+        const dataFile: Item = new Item(file.id, file.name, file.updated, file.created,
+          file.size, file.mediaLink, file.path, this.directory,
+          Type.DOWNLOAD, ItemStatus.PENDING, '', '', '', this.preserveStructure);
```

Comment:
I think you can improve this slightly more by making a constructor specific to file upload/download that omits the arguments `Type.DOWNLOAD, ItemStatus.PENDING, '', '', ''` since they are the same in both cases. Not sure if you could replace the `file` arguments as well (e.g. pass `file` for download and pass `file.data` for upload, but that would help too.

Reply:
We can't create multiple constructors for each type because multiple implementations are not allowed. It would be nice to send only one parameter with Item argument but we are using the same constructor to create   files, and folders in Bucket Service, in this case we don't have the specific item so we have to create it before call the constructor.

Reply:
OK